commit 592e71de516a590c9bf8b00448a050a46aac7df0
Author: Michael Fellinger <michael.fellinger@iohk.io>
Date:   Fri May 13 14:09:43 2022 +0200

    avoid assuming order of extras

diff --git a/srvaddr.go b/srvaddr.go
index d153f9b..913d689 100644
--- a/srvaddr.go
+++ b/srvaddr.go
@@ -141,13 +141,17 @@ func main() {
 				if dns.IsFqdn(host) {
 					host = host[:len(host)-1]
 				}
-				srvs[i] = service{Host: host, FQDN: srv.Target, Port: srv.Port}
-			}
-		}
 
-		for i, extra := range in.Extra {
-			if rec, ok := extra.(*dns.A); ok {
-				srvs[i].IP = rec.A
+				var ip net.IP
+				for _, extra := range in.Extra {
+					if rec, ok := extra.(*dns.A); ok {
+						if rec.Header().Name == srv.Target {
+							ip = rec.A
+						}
+					}
+				}
+
+				srvs[i] = service{Host: host, FQDN: srv.Target, Port: srv.Port, IP: ip}
 			}
 		}
 
